src/SFML/Graphics/Texture.cpp

```diff
@@ -350,6 +350,22 @@ Image Texture::copyToImage() const
         glCheck(GLEXT_glDeleteFramebuffers(1, &frameBuffer));
 
         glCheck(GLEXT_glBindFramebuffer(GLEXT_GL_FRAMEBUFFER, previousFrameBuffer));
+
+        if (m_pixelsFlipped)
+        {
+            // Flip the texture vertically
+            const int stride        = static_cast<int>(m_size.x * 4);
+            Uint8*    currentRowPtr = pixels.data();
+            Uint8*    nextRowPtr    = pixels.data() + stride;
+            Uint8*    reverseRowPtr = pixels.data() + (stride * static_cast<int>(m_size.y - 1));
+            for (unsigned int y = 0; y < m_size.y / 2; ++y)
+            {
+                std::swap_ranges(currentRowPtr, nextRowPtr, reverseRowPtr);
```

Comment:
[`std::swap_ranges`](https://en.cppreference.com/w/cpp/algorithm/swap_ranges) is a C++17 feature. We need a C++03-compliant solution for SFML 2.

Reply:
swap_ranges already existed before, it's only the second variant of that cppreference.com page that was added in c++17

Reply:
😅 My bad!